modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -742,7 +766,8 @@ public boolean onUpdate(@NotNull WatchEvent evt) {
                         }
 
                         try {
-                            assert evt.single();
+                            assert evt.single() : evt.entryEvents().stream()
+                                    .map(entry -> entry.newEntry() == null ? "null" : entry.newEntry().key()).collect(toList());
```

Comment:
There are still some places where you do expect table.zoneId to be null, which is not possible, e.g.
return tableZoneId != null && tableZoneId.equals(zoneView.zoneId());
It's however not valid, so you can add assert if you wan't or just remove tableZoneId != null
Please fix wherever needed.

Reply:
I found the only one place. Fixed.